src/st/dre/succubus.c

```diff
@@ -0,0 +1,958 @@
+/*
+ * Overlay: DRE
+ * Enemy: Succubus Boss
+ * ID: 0x19
+ * BOSS ID: 9
+ */
+
+#include "dre.h"
+
+typedef enum {
+    /* 0 */ SUCCUBUS_INIT,
+    /* 1 */ SUCCUBUS_CS_1,
+    /* 2 */ SUCCUBUS_CS_2,
+    /* 3 */ SUCCUBUS_CS_3,
+    /* 4 */ SUCCUBUS_CS_4,
+    /* 5 */ SUCCUBUS_IDLE,
+    /* 7 */ SUCCUBUS_FLY = 7,
+    /* 8 */ SUCCUBUS_PETAL_ATTACK,
+    /* 9 */ SUCCUBUS_CHARGE,
+    /* 11 */ SUCCUBUS_CLONE_ATTACK = 11,
+    /* 12 */ SUCCUBUS_SPIKE_ATTACK,
+    /* 13 */ SUCCUBUS_TAUNT,
+    /* 14 */ SUCCUBUS_GET_HIT,
+    /* 15 */ SUCCUBUS_FACE_PLAYER,
+    /* 16 */ SUCCUBUS_NEXT_ACTION_CHECK,
+    /* 18 */ SUCCUBUS_DYING = 18,
+    /* 255 */ SUCCUBUS_DEBUG = 255,
```

Comment:
are the comments necessary? Especially now that the function is fully decompiled and some enum values have their value explicitly set. Same question applies for all the enums below

Reply:
They are very useful still, let me explain why:
Even though the function is decompiled, there's still much to document in it, the way i did this was a combination of looking at the code and playing with the debugger to force the enemy to do whatever step i needed and watch it's behaviour. Having those comments there helped me out so much without having to figure out the numbers of steps that i already documented. The same is true for the next person who comes to document what's left of it, for the person that comes to decompile the function from another version (there could be differences, specially now that we found a bug in it), or the person who comes to make a mod of it, having the comments there is not invasive and it just helps in a lot of ways.